src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionTests_WithElement_Constructor.cs

```diff
@@ -0,0 +1,1365 @@
+                    Console.WriteLine($"{list.Values.Count()}");
+                }
+            }
+            """;
+
+        CompileAndVerify(source, expectedOutput: IncludeExpectedOutput("3"));
+    }
+
+    #endregion
+
+    #region Dynamic Tests
+
+    [Fact]
+    public void WithElement_DynamicArguments()
+    {
+        var source = """
+            using System;
+            using System.Collections.Generic;
+            
+            class MyList<T> : List<T>
+            {
+                public object Value { get; }
+                
+                public MyList(object value) : base()
+                {
+                    Value = value;
+                }
+            }
+            
+            class C
+            {
+                static void Main()
+                {
+                    dynamic d = 42;
+                    MyList<int> list = [with(d), 1];
+                    Console.WriteLine(list.Value);
+                }
+            }
+            """;
+
+        CreateCompilation(source, references: [CSharpRef]).VerifyDiagnostics(
+            // (19,34): error CS9337: Collection arguments cannot be dynamic
+            //         MyList<int> list = [with(d), 1];
+            Diagnostic(ErrorCode.ERR_CollectionArgumentsDynamicBinding, "d").WithLocation(19, 34));
+    }
+
+    [Fact]
+    public void WithElement_DynamicNamedArguments()
+    {
+        var source = """
+            using System.Collections.Generic;
+            
+            class MyList<T> : List<T>
+            {
+                public MyList(object value) : base() { }
+            }
+            
+            class C
+            {
+                void M()
+                {
+                    dynamic d = 42;
+                    MyList<int> list = [with(value: d)];
+                }
+            }
+            """;
+
+        CreateCompilation(source).VerifyDiagnostics(
+            // (13,41): error CS9337: Collection arguments cannot be dynamic
+            //         MyList<int> list = [with(value: d)];
+            Diagnostic(ErrorCode.ERR_CollectionArgumentsDynamicBinding, "d").WithLocation(13, 41));
+    }
+
+    [Fact]
+    public void WithElement_DynamicType()
+    {
+        var source = """
+            using System.Collections.Generic;
+            
+            class C
+            {
+                void M()
+                {
+                    dynamic d = new List<int>();
+                    var list = [with(capacity: 10)] as dynamic;
+                }
+            }
+            """;
+
+        CreateCompilation(source).VerifyDiagnostics(
+            // (8,20): error CS9176: There is no target type for the collection expression.
+            //         var list = [with(capacity: 10)] as dynamic;
+            Diagnostic(ErrorCode.ERR_CollectionExpressionNoTargetType, "[with(capacity: 10)]").WithLocation(8, 20));
+    }
+
+    #endregion
+
+    #region ArgList Tests
+
+    [ConditionalFact(typeof(WindowsOnly))]
```

Comment:
Do not make these `ConditionalFact`s. The test can run on all platforms.

Reply:
aa23cbd6d1c